test/test_algorithm.jl

```diff
@@ -1043,3 +1043,77 @@ end
     @test isapprox(objective_value(m), 13; atol = 1e-4)
     @test MOI.get(m, Alpine.NumberOfIterations()) == 0
 end
+
+@testset "Test integer variable support via IntegerToZeroOneBridge" begin
+    test_solver = optimizer_with_attributes(
+        Alpine.Optimizer,
+        "nlp_solver" => IPOPT,
+        "mip_solver" => HIGHS,
+        "minlp_solver" => JUNIPER,
+    )
+
+    m = JuMP.Model(test_solver)
+
+    @variable(m, -10 <= objvar <= 20, Int)
+
+    @objective(m, Min, objvar)
+
+    JuMP.optimize!(m)
+
+    @test termination_status(m) == MOI.OPTIMAL
+    @test isapprox(objective_value(m), -10; atol = 1e-4)
+    @test isapprox(value(objvar), -10, atol = 1E-6)
+    @test MOI.get(m, Alpine.NumberOfIterations()) == 0
+end
+@testset "Test integer variable support 0" begin
+    test_solver = optimizer_with_attributes(
+        Alpine.Optimizer,
+        "nlp_solver" => IPOPT,
+        "mip_solver" => HIGHS,
+        "minlp_solver" => JUNIPER,
+    )
+
+    m = JuMP.Model(test_solver)
+
+    @variable(m, objvar, Int)
+    @constraint(m, -10 <= objvar)
```

Comment:
So only the bounds added when creating the variable are considered bounds.
Is this by design, or am i missing some toggle?

Reply:
You are right! As it is, there is no toggle to convert a constraint-based bound into a variable bound. But, after the bound propagation algorithm with `presolve_bt = true`, a new model is created with deduced bounds on the variables based on constraints. In this sense, this toggle seemed unnecessary. But do you think it would be necessary in some cases?

Reply:
> You are right! As it is, there is no toggle to convert a constraint-based bound into a variable bound.

Ok, then this is basically ready for the review then.

> But, after the bound propagation algorithm with `presolve_bt = true`, a new model is created with deduced bounds on the variables based on constraints. In this sense, this toggle seemed unnecessary. But do you think it would be necessary in some cases?

Right. But all that happens after `IntegerToZeroOneBridge` bridge runs
and encounters variables whose bounds may be specified as constraints,
which means those variables may not have bounds after all,
that are required for `IntegerToZeroOneBridge` to work in the first place,
which in turn means the whole model still fails to be supported.

Reply:
Agree! Before this bridge fails, we could leave this to the user and include an explicit error message: "If known, include variable bounds explicitly instead of constraints" ?

Reply:
As tests show, `IntegerToZeroOneBridge` already produces an error message,
so if anything, said error message should/could be improved on MOI side.
If that wasn't the question, i did not follow what the question was.

Reply:
This is not a bound. This is the constraint `1.0 * objvar in GreaterThan(-10.0)`

Reply:
You need `@variable(m, -10 <= objvar <= 20, Int)`, or `set_lower_bound(objvar, -10)`.

Reply:
Right. At this point those are more of a negative tests to demonstrate
that such syntax does not produce bounds needed for this to work.